internal/internal_worker.go

```diff
@@ -875,7 +875,7 @@ func (aw *aggregatedWorker) Start() error {
 }
 
 var binaryChecksum string
-var binaryChecksumLock sync.Mutex
+var binaryChecksumLock sync.RWMutex
```

Comment:
this does seem better, but eh... if we're going to rewrite this, let's just use a `sync.Once`.  the semantics are much more accurate and it's probably faster.

Reply:
There's `SetBinaryChecksum()` function so can't use `sync.Once` as I mentioned in PR description.

Reply:
ah, it is fully mutable isn't it.  I was probably thinking of the sticky cache size.

atomics are probably fine then tbh, doesn't seem like anything concurrently starting-and-updating-checksum can expect any sort of ordering between calls.

but yea, rwmutex is an improvement either way, I can 👍 the PR for that.